MetadataExtractor/Formats/Png/PngChunkReader.cs

```diff
@@ -30,14 +30,14 @@
 namespace MetadataExtractor.Formats.Png
 {
     /// <author>Drew Noakes https://drewnoakes.com</author>
-    public sealed class PngChunkReader
+    public static class PngChunkReader
     {
         private static readonly byte[] _pngSignatureBytes = { 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A };
 
         /// <exception cref="PngProcessingException"/>
         /// <exception cref="System.IO.IOException"/>
         [NotNull]
-        public IEnumerable<PngChunk> Extract([NotNull] SequentialReader reader, [CanBeNull] ICollection<PngChunkType> desiredChunkTypes)
+        public static IEnumerable<PngChunk> Extract([NotNull] SequentialReader reader, [CanBeNull] ICollection<PngChunkType> desiredChunkTypes)
```

Comment:
Making all these Extract methods static means they will always be loaded at startup - even if the consumer doesn't need it. For example, if I'm not extracting on a PNG file, it will still load this code the first time the library is created.

Making it static would probably help overall efficiency if you're re-using the same library instance against many files in a loop. But I wonder if it's better to do lazy loading instead.

Reply:
@kwhopper Are you thinking of static constructors?

Reply:
Not really. More of a local, initially null static variable in each MetadataReader class that is set the first time it is needed. After that, it would be reused - like a lazy loaded property.

However, that implies each of the MetadataReader classes are either all static already or only created one time from the top. That would have to be checked. Otherwise this might not be effective.

Reply:
Static fields and constructors are initialized before their first use. These readers have neither.

I still don't understand what is loaded at runtime.